diff --git a/node_modules/@rc-component/portal/es/Portal.js b/node_modules/@rc-component/portal/es/Portal.js
index 734b24b..f9c77f2 100644
--- a/node_modules/@rc-component/portal/es/Portal.js
+++ b/node_modules/@rc-component/portal/es/Portal.js
@@ -54,11 +54,11 @@ const Portal = /*#__PURE__*/React.forwardRef((props, ref) => {
     const customizeContainer = getPortalContainer(getContainer);
 
     // Tell component that we check this in effect which is safe to be `null`
-    setInnerContainer(() =>
-    // React do the state update even the value is the same,
-    // Use function call to force React to compare update
-    customizeContainer ?? null);
-  });
+    setInnerContainer(prev => {
+      const next = customizeContainer ?? null;
+      return prev === next ? prev : next;
+    });
+  }, [getContainer]);
   const [defaultContainer, queueCreate] = useDom(mergedRender && !innerContainer, debug);
   const mergedContainer = innerContainer ?? defaultContainer;
 
diff --git a/node_modules/@rc-component/portal/lib/Portal.js b/node_modules/@rc-component/portal/lib/Portal.js
index 258a296..687b071 100644
--- a/node_modules/@rc-component/portal/lib/Portal.js
+++ b/node_modules/@rc-component/portal/lib/Portal.js
@@ -63,11 +63,11 @@ const Portal = /*#__PURE__*/React.forwardRef((props, ref) => {
     const customizeContainer = getPortalContainer(getContainer);
 
     // Tell component that we check this in effect which is safe to be `null`
-    setInnerContainer(() =>
-    // React do the state update even the value is the same,
-    // Use function call to force React to compare update
-    customizeContainer ?? null);
-  });
+    setInnerContainer(prev => {
+      const next = customizeContainer ?? null;
+      return prev === next ? prev : next;
+    });
+  }, [getContainer]);
   const [defaultContainer, queueCreate] = (0, _useDom.default)(mergedRender && !innerContainer, debug);
   const mergedContainer = innerContainer ?? defaultContainer;
 
